Quran/SQLiteSearchableAyahTextPersistence.swift

```diff
@@ -39,22 +69,33 @@ class SQLiteSearchableAyahTextPersistence: ReadonlySQLitePersistence, Searchable
     private func search(for term: String, connection: Connection) throws -> [SearchResult] {
         CLog("Search for:", term)
         let searchTerm = cleanup(term: term)
+        let arabicSearchTerm = searchTerm.replacingOccurrences(of: regex,
+                                                               with: "_",
+                                                               options: .regularExpression)
+        let pattern = generateRegexPattern(term: term)
         let query = Database.table
-            .select(Database.Columns.snippet, Database.Columns.sura, Database.Columns.ayah)
-            .filter(Database.Columns.text.match("\(searchTerm)"))
-            .limit(300)
+            .select(Database.Columns.text, Database.Columns.sura, Database.Columns.ayah)
+            .filter(Database.Columns.text.like("%\(arabicSearchTerm)%"))
```

Comment:
this no longer uses `match` because it takes advantage of the `_` character in `like`, which matches any character. essentially, we replace any of the characters that could have valid other options (ا for example) with an `_` to match anything. the postprocessing step them ensures that whatever filled that `_` is what we want.

sadly, that also means we have to highlight ourselves.